File: 09-29/Cadena de Responsabilidad/CoR_java/Banco.java
package CoR_java;

public class Banco implements IAprobador{
    private IAprobador next;

    public Banco() {
    }

    public Banco(IAprobador next) {
        this.next = next;
    }

    @Override
    public IAprobador getNext() {
        return next;
    }
    // ejecutivo - lider - gerente - director
    @Override
    public void solicitudPrestamo(int monto) {
        EjecutivoCuenta ejecutivo = new EjecutivoCuenta();
        this.setNext(ejecutivo);
        LiderTeamEjecutivo lider = new LiderTeamEjecutivo();
        ejecutivo.setNext(lider);
        Gerente gerente = new Gerente();
        lider.setNext(gerente);
        Director director = new Director();
        gerente.setNext(director);

        next.solicitudPrestamo(monto);
    }

    @Override
    public void setNext(IAprobador aprobador) {
        next = aprobador;
    }
}

File: 09-29/Cadena de Responsabilidad/CoR_python/banco.py
from personas import *
import personas

# ejecutivo - lider - gerente - director


class Banco():
    director = Director()
    gerente = Gerente()
    lider = LiderTeamEjecutivo()
    ejecutivo = EjecutivoCuenta()

    director.set_next(gerente).set_next(lider).set_next(ejecutivo)
    personas.main(director)
